### PR TITLE
Attempted fix to issue #63551 

### PR DESCRIPTION
**Pull request by TheOregonTrail**
Well this is my first PR so hopefully I didn't do too bad.

As said I'm attempting to fix issue #63551 and remove the unnecessary check for is_unit.

would recommend removing `let is_unit = input_ty.is_unit()` unless there is another use.

### CODE REVIEW

**Comment by bjorn3** [comment]
Could you please remove the rustfmt commit? While it is ok to format the lines you are working with, fornatting the rest gives too much noise.

**Comment by petrochenkov** [comment]
r? @bjorn3 
cc @eddyb

**Comment by petrochenkov** [comment]
@bors delegate=bjorn3
(Not sure all assignable people have r+ rights.)

**Comment by bjorn3** [comment]
@petrochenkov I am not part of the rust team.

r? @eddyb (does that work?)

File: src/librustc_codegen_ssa/mir/rvalue.rs [hunk 1]
```diff
@@ -1,42 +1,41 @@
...
-use crate::common::{self, RealPredicate, IntPredicate};
+use crate::common::{self, IntPredicate, RealPredicate};
+use crate::MemFlags;
 
 use crate::traits::*;
 
-use super::{FunctionCx, LocalRef};
 use super::operand::{OperandRef, OperandValue};
 use super::place::PlaceRef;
+use super::{FunctionCx, LocalRef};
```
Comment (eddyb) [inline src/librustc_codegen_ssa/mir/rvalue.rs]: Please do not run `rustfmt`, it makes the actual changes harder to review.

**Comment by petrochenkov** [comment]
@bjorn3 
>I am not part of the rust team.

(That's not too important, I know that you worked on `rustc_codegen_ssa` and familiar with the code, so you can review.)

**Comment by bjorn3** [comment]
> (That's not too important, I know that you worked on rustc_codegen_ssa and familiar with the code, so you can review.)

Didn't know that. Is that written down somewhere as policy?

File: src/librustc_codegen_ssa/mir/rvalue.rs [hunk 2]
```diff
@@ -546,7 +546,8 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
     ) -> Bx::Value {
         let is_float = input_ty.is_floating_point();
         let is_signed = input_ty.is_signed();
-        let is_unit = input_ty.is_unit();
+        // No need for this anymore ISSUE:#63551
+        let _is_unit = input_ty.is_unit();
```
Comment (bjorn3) [inline src/librustc_codegen_ssa/mir/rvalue.rs]: Please remove this line and the one above.

File: src/librustc_codegen_ssa/mir/rvalue.rs [hunk 3]
```diff
@@ -584,13 +585,8 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
...
             mir::BinOp::Ne | mir::BinOp::Lt | mir::BinOp::Gt |
-            mir::BinOp::Eq | mir::BinOp::Le | mir::BinOp::Ge => if is_unit {
-                bx.cx().const_bool(match op {
-                    mir::BinOp::Ne | mir::BinOp::Lt | mir::BinOp::Gt => false,
-                    mir::BinOp::Eq | mir::BinOp::Le | mir::BinOp::Ge => true,
-                    _ => unreachable!()
-                })
-            } else if is_float {
+            mir::BinOp::Eq | mir::BinOp::Le | mir::BinOp::Ge => {
+              if is_float {
```
Comment (bjorn3) [inline src/librustc_codegen_ssa/mir/rvalue.rs]: Please remove the `{` on the line above and merge this line into the one above.

File: src/librustc_codegen_ssa/mir/rvalue.rs [hunk 4]
```diff
@@ -600,10 +596,12 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                     base::bin_op_to_icmp_predicate(op.to_hir_binop(), is_signed),
                     lhs, rhs
                 )
+              }
```
Comment (bjorn3) [inline src/librustc_codegen_ssa/mir/rvalue.rs]: Please remove this line.

File: src/librustc_codegen_ssa/mir/rvalue.rs [hunk 5]
```diff
@@ -600,10 +596,12 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
...
             }
         }
     }
 
+
```
Comment (bjorn3) [inline src/librustc_codegen_ssa/mir/rvalue.rs]: Please remove this newline.

**Comment by bjorn3** [comment]
Please squash your comments after fixing the review comments above. See for example http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html for how to squash.

**Comment by luigishat** [comment]
Sorry Having trouble with git rebase I have a merge conflict with rvalue.rs and It's giving me a headache.

**Comment by luigishat** [comment]
@bjorn3 I tried to git squash it but I think I messed it up. When I finish with git rebase I keep getting thrown back a commit and am required to pull and henceforth get stuck pushing two commits at once ("Added two commits six days ago"). Help would be appreciated

**Comment by bjorn3** [comment]
Maybe reset your branch back to master and perform you changes again.

```bash
$ git fetch origin # get latest version of rust
$ git reset --hard origin/master # reset current branch back to latest version of rust (throws all changes away)
$ # perform changes and commit
$ git push --force # push your changes to your fork
```

**Comment by JohnCSimon** [comment]
Ping from triage:
@luigishat can you please address the comments here? thanks!

**Comment by luigishat** [comment]
@JohnCSimon @bjorn3 Sorry about the late action I've had problems with my computer and just got my self repairs done today. Thanks for being patient :+1:

**Review by bjorn3 — APPROVED** [review]

**Comment by luigishat** [comment]
So what now @bjorn3 ?

**Comment by luigishat** [comment]
@bjorn3 @Kixiron could you try r+ because right now I think this PR is stagnant.

**Comment by bjorn3** [comment]
I can try

@bors r+

**Comment by bjorn3** [comment]
@luigishat It seems some tests changed their error order. Please bless them. (`./x.py test --bless`)

**Comment by bjorn3** [comment]
I should have checked the PR ci run before `r+`ing this.

**Comment by luigishat** [comment]
@bjorn3  Do you know why it says Alex Crichton canceled the build ?

**Comment by bjorn3** [comment]
No, where are you seeing that?

**Comment by mati865** [comment]
> Do you know why it says Alex Crichton canceled the build ?

When one of the builds fails bot cancels other jobs to save time by letting other jobs run.
I think Alex configured this using his personal account and that's why his name appears. It doesn't matter since this is legitimate failure.

**Comment by luigishat** [comment]
@bjorn3 x.py test --bless leaves me with 82 errors concerning for the test suite [debuginfo-gdb+lldb]. I'm also getting "error: gdb failed to execute". What is gdb's role in the compiler ?

**Comment by bjorn3** [comment]
gdb is used in tests testing debuginfo. It should be fine to ignore those tests.

**Comment by luigishat** [comment]
@bjorn3 what should I look for in failed-doctest-output.rs ?

**Comment by bjorn3** [comment]
Running `./x.py --bless src/test/rustdoc-ui` should fix all rustdoc tests, while skipping the tests requiring gdb.

**Comment by luigishat** [comment]
Ok I ran ./x.py test --stage 1 src/test/rustdoc-ui --bless. The Tests all passed but I can't add anything to a new commit because nothing changed. @bjorn3

**Comment by bjorn3** [comment]
Maybe try rebasing this PR. If CI succeeds, this PR will be ready to go.

**Comment by JohnCSimon** [comment]
Ping from triage - this hasn't been touched in 5 days.
@luigishat @bjorn3 seems that CI has failed 
Thanks.

**Comment by luigishat** [comment]
@bjorn3 Could we try testing this commit with whatever we did before ?

**Comment by luigishat** [comment]
@bjorn3 If you have the time could you please fix it and tell me what you did ? Thanks so much for your patience you really have been a big help stepping me through this.

**Comment by bjorn3** [comment]
I decided to make a new branch with your changes:

```bash
$ git checkout master
$ git pull # Get latest changes
$ git checkout -b fix_63906 # Make new branch to apply changes

# Fetch all your commits
$ git remote add luigishat https://github.com/luigishat/rust.git
$ git fetch luigishat

# Take the commit with the actual changes and apply it to the current branch
$ git cherry-pick b139849bf1842c7cf7ea0cc78e3ddc362cb2ce02

# Change the commit message to something nicer
$ git commit --amend -m "Remove unreachable unit tuple compare binop codegen"

$ git push -u my fix_63906 # Push branch to my fork
```

Normally you would do something like:

```
$ git fetch origin # Assuming origin refers to https://github.com/rust-lang/rust
$ git rebase origin/master
$ git push --force-with-lease # Push rebased version (--force-with-lease instead of --force prevents forcing when your fork changed between the last time you pulled and the push)
```

**Pull request closed without merging**